src/server/shared/DataStores/DBCDatabaseLoader.cpp

```diff
@@ -0,0 +1,148 @@
+ *
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "DBCDatabaseLoader.h"
+#include "Common.h"
+#include "DatabaseEnv.h"
+#include "Errors.h"
+#include "Log.h"
+#include "StringFormat.h"
+#include <sstream>
+
+DBCDatabaseLoader::DBCDatabaseLoader(char const* tableName, char const* dbcFormatString, std::vector<char*>& stringPool)
+    : _sqlTableName(tableName),
+    _dbcFormat(dbcFormatString),
+    _sqlIndexPos(0),
+    _recordSize(0),
+    _stringPool(stringPool)
+{
+    // Get sql index position
+    int32 indexPos = -1;
+    _recordSize = DBCFileLoader::GetFormatRecordSize(_dbcFormat, &indexPos);
+
+    ASSERT(_recordSize);
+}
+
+char* DBCDatabaseLoader::Load(uint32& records, char**& indexTable)
+{
+    std::string query = acore::StringFormat("SELECT * FROM `%s` ORDER BY `ID` DESC", _sqlTableName);
+
+    // no error if empty set
+    QueryResult result = WorldDatabase.Query(query.c_str());
+    if (!result)
+        return nullptr;
+
+    // Check if sql index pos is valid
+    if (int32(result->GetFieldCount() - 1) < _sqlIndexPos)
+    {
+        ASSERT(false, "Invalid index pos for dbc: '%s'", _sqlTableName);
+        return nullptr;
+    }
+
+    // Resize index table
+    // database query *MUST* contain ORDER BY `index_field` DESC clause
+    uint32 indexTableSize = std::max(records, (*result)[_sqlIndexPos].GetUInt32() + 1);
+    if (indexTableSize > records)
+    {
+        char** tmpIdxTable = new char*[indexTableSize];
+        memset(tmpIdxTable, 0, indexTableSize * sizeof(char*));
+        memcpy(tmpIdxTable, indexTable, records * sizeof(char*));
+        delete[] indexTable;
+        indexTable = tmpIdxTable;
+    }
+
+    std::unique_ptr<char[]> dataTable = std::make_unique<char[]>(result->GetRowCount() * _recordSize);
+    std::unique_ptr<uint32[]> newIndexes = std::make_unique<uint32[]>(result->GetRowCount());
+    uint32 newRecords = 0;
+
+    // Insert sql data into the data array
+    do
+    {
+        Field* fields = result->Fetch();
+        uint32 indexValue = fields[_sqlIndexPos].GetUInt32();
+        char* dataValue = indexTable[indexValue];
```

Comment:
From our discord chat:

> You're gettingh the dataValue from the indexTable and if the sql exists, then you override it
> you can do it only if dbc exist, but sql should override/create a value inside the indexTable in any case. If previous value exists, sql will override it, if not previous value exists, sql will create it
> in case of no dbc files installed, ofc it will always the last condition and sql will be the only loader to fill the indexTable